apis/kueue/webhooks/clusterqueue_webhook.go

```diff
@@ -99,39 +99,50 @@ func (w *ClusterQueueWebhook) ValidateDelete(ctx context.Context, obj runtime.Ob
 func ValidateClusterQueue(cq *kueue.ClusterQueue) field.ErrorList {
 	path := field.NewPath("spec")
 
-	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, ValidateResources(cq, path.Child("resources"))...)
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateResources(cq.Spec.Resources, path.Child("resources"))...)
 	allErrs = append(allErrs, validateQueueingStrategy(string(cq.Spec.QueueingStrategy), path.Child("queueingStrategy"))...)
 	allErrs = append(allErrs, validateNamespaceSelector(cq.Spec.NamespaceSelector, path.Child("namespaceSelector"))...)
 
 	return allErrs
 }
 
-func ValidateResources(cq *kueue.ClusterQueue, path *field.Path) field.ErrorList {
-	allErrs := field.ErrorList{}
+func validateResources(resources []kueue.Resource, path *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	flavorsPerRes := make([]sets.String, len(resources))
 
-	for i, resource := range cq.Spec.Resources {
-		allErrs = append(allErrs, validateResourceName(string(resource.Name), path.Index(i).Child("name"))...)
+	for i, resource := range resources {
+		path := path.Index(i)
+		allErrs = append(allErrs, validateResourceName(string(resource.Name), path.Child("name"))...)
 
-		flavorPath := path.Index(i).Child("flavors")
+		flavorsPerRes[i] = make(sets.String, len(resource.Flavors))
 		for j, flavor := range resource.Flavors {
-			allErrs = append(allErrs, validateFlavorName(string(flavor.Name), flavorPath.Index(j).Child("name"))...)
-			allErrs = append(allErrs, validateFlavorQuota(flavor, flavorPath.Index(j).Child("quota"))...)
+			path := path.Child("flavors").Index(j)
+			allErrs = append(allErrs, validateFlavorName(string(flavor.Name), path.Child("name"))...)
+			allErrs = append(allErrs, validateFlavorQuota(flavor, path.Child("quota"))...)
+			flavorsPerRes[i].Insert(string(flavor.Name))
+		}
+		for j := 0; j < i; j++ {
+			if !flavorsPerRes[i].HasAny(flavorsPerRes[j].UnsortedList()...) || matchesFlavorsInOrder(resource.Flavors, resources[j].Flavors) {
+				continue
+			}
+			err := field.Invalid(path.Child("flavors"), resource.Flavors, fmt.Sprintf("has flavors present in resource %s; all flavors must be different or they all must be present in the same order", resources[j].Name))
```

Comment:
would the name of `resources[i]` show up in the message or just its index?

Reply:
just the index